src/Controllers/LoginSessionController.php

```diff
@@ -29,50 +33,52 @@ class LoginSessionController extends Controller
      */
     public function remove(HTTPRequest $request): HTTPResponse
     {
-        return $this->removeLoginSession($request);
-    }
-
-    private function removeLoginSession(HTTPRequest $request): HTTPResponse
-    {
-        $failureMessage = _t(__CLASS__ . '.REMOVE_FAILURE', 'Something went wrong.');
-        try {
-            // Ensure CSRF protection
-            if (!SecurityToken::inst()->checkRequest($request)) {
-                return $this->jsonResponse([
-                    'success' => false,
-                    'message' => $failureMessage
-                ]);
-            }
+        if (empty(Security::getCurrentUser())) {
```

Comment:
Isn't there some sort of protection on /admin that happens before this point?  Or is that simply to customise the message in this specific scenario only?

Seems like this "your session has expired" functionality should live somehwhere more global?

Reply:
It's a regular controller now, so it's no longer showing under `admin`. I guess we could put a restriction on the action, but we wouldn't got a nice error message in that scenario.